checkov/cloudformation/cfn_utils.py

```diff
@@ -236,3 +238,77 @@ def validate_properties_in_resources_are_dict(template: dict[str, Any]) -> bool:
         if 'Properties' in resource and not isinstance(resource['Properties'], dict) or "." in resource_name:
             return False
     return True
+
+
+def enrich_resources_with_globals(original_template: dict[str, Any]) -> dict[str, Any]:
+    """
+    Creates a new CloudFormation template dictionary with global properties applied to the resources.
+    :param original_template: The parsed CloudFormation template as a dictionary.
+    :return: A new CloudFormation template with enriched resources.
+    """
+
+    new_template = pickle_deepcopy(original_template)  # Create a deep copy of the original template
+
+    try:
+        # Check if Globals exist in the template
+        global_props = new_template.get('Globals', {})
+
+        supported_types = ['Api', 'Function', 'HttpApi', 'SimpleTable', 'StateMachine']
+
+        # Supported AWS serverless type mappings to their corresponding Globals
+        supported_types_and_globals = {f"AWS::Serverless::{type}": global_props.get(type, {}) for type in supported_types}
+
+        # Iterate over the resources in the template copy
+        for _resource_name, resource_details in new_template.get('Resources', {}).items():
+            resource_type = resource_details.get('Type', '')
+            if (resource_type not in supported_types_and_globals):
+                continue
+            global_properties = supported_types_and_globals.get(resource_type, {})
+            resource_properties = resource_details.setdefault('Properties', {})
+            skip_properties = ['Tags']
+            for property in skip_properties:
+                global_properties.pop(property, None)
+
+            merged_properties = deep_merge(resource_properties, global_properties)
+
+            # Set the merged properties back into the resource details
+            resource_details['Properties'] = merged_properties
+
+    except Exception as e:
+        logging.warning(f"Failed to create a new template with enriched resources: {e}")
+        return original_template
+
+    return new_template  # Return the new template even if there were no globals to apply
+
+
+def deep_merge(dict1: DictNode, dict2: DictNode) -> DictNode:
```

Comment:
maybe this function can be moved under `DictNode` class?

Reply:
Sure